packages/plugin-ma-tracking/src/advanced-campaign-tracker.ts

```diff
@@ -0,0 +1,43 @@
+import { Campaign, CampaignTrackerOptions } from '@amplitude/analytics-types';
+import { CampaignTracker } from '@amplitude/analytics-client-common';
+
+export class AdvancedCampaignTracker extends CampaignTracker {
+  resetSessionOnNewCampaign: boolean;
+
+  private _currentCampaign!: Campaign;
```

Comment:
No ! please

Reply:
This value is created while calling `refreshCampaignState` function, but the linter complains about it. `!` is used to declare it is not undefined.

Reply:
compiler or linter is correct when it called out that this property can be undefined. we should add a guard when accessing, instead of bypassing type checks

Reply:
Added a default value